include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -424,16 +424,128 @@ struct tuple<T1, T...>
         return *this;
     }
 
-    friend bool
-    operator==(const tuple& __lhs, const tuple& __rhs)
+    // The following operators enable support of binary comparison of oneDPL internal tuples with other oneDPL internal
+    // tuples with the same number of types where individual elements are binary comparable
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator==(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
     {
         return __lhs.holder.value == __rhs.holder.value && __lhs.next == __rhs.next;
     }
-    friend bool
-    operator!=(const tuple& __lhs, const tuple& __rhs)
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator!=(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
+    {
+        return !(__lhs == __rhs);
+    }
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator<(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
+    {
+        return __lhs.holder.value < __rhs.holder.value ||
+               (!(__rhs.holder.value < __lhs.holder.value) && __lhs.next < __rhs.next);
+    }
+
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator<=(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
+    {
+        return !(__rhs < __lhs);
+    }
+
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator>(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
+    {
+        return __rhs < __lhs;
+    }
+
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator>=(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U...>& __rhs)
+    {
+        return !(__lhs < __rhs);
+    }
+
+    // The following operators enable support of binary comparison of oneDPL internal tuples with std::tuple as rhs
+    // with the same number of types where individual elements are binary comparable
+    template <typename... _U, typename = std::enable_if_t<sizeof...(_U) == (sizeof...(T) + 1)>>
+    friend constexpr bool
+    operator==(const tuple& __lhs, const std::tuple<_U...>& __rhs)
+    {
+        return __lhs.holder.value == std::get<0>(__rhs) && __lhs.next == oneapi::dpl::__internal::get_tuple_tail(__rhs);
```

Comment:
Looks like here we will make copying of the rest of our tuple too...

Reply:
Yes, you are correct.  Working on a fix...

Reply:
OK I believe we should be free from copies now.